### ISSUE TITLE
Voltage and current errors

### ISSUE THREAD

**Issue by viktor-ktorvi** [viktor-ktorvi/mlpf#2]
TODO:
- [x] bounds. voltage bounds follow and I won't be making a dedicated function for it
- [ ] current
For voltage something like:

```python
def upper_voltage_error(voltage_magnitude, voltage_max):
    return torch.max(0, voltage_magnitude - voltage_max)

def lower_voltage_error(voltage_magnitude, voltage_min):
    return torch.min(0, voltage_magnitude - voltage_min)
```

Notice that these are just errors and can be negative(for lower_voltage_error). To get a loss someone would make a loss module where they abs and scalarize.

```python
def current(source, target, voltage_magnitudes, angles, conductances, susceptances):
    y = conductances + 1j * susceptances

    complex_voltages = voltage_magnitudes * torch.exp(1j*angles)
   
    # wait, this will give 0 for the shunt current. Maybe add a mask source == target put v_ii * y_ii where true
    return (complex_voltages[source] - complex_voltages[target]) * y
```

-[ ] Test the current for equality with pypowers current i.e. makeYbus will return some Y_to and Y_from which when multiplied with voltages will yield currents.

**Comment by viktor-ktorvi** [comment]
Probably not gonna implement the current errors except maybe straight for ppcs as in current_errors(ppcs)
